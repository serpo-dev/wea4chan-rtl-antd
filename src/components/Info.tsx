import { FC } from "react";
import { Link } from "react-router-dom";

export const Info: FC = () => {
    return (
        <div
            style={{
                fontSize: "16px",
                textAlign: "center",
                fontWeight: 500,
                padding: "15px",
            }}
        >
            <Link
                to="/about"
                data-testid="about_link"
            >
                About
            </Link>
            <span> | </span>
            <a
                href="https://github.com/yphwd"
                target="_blank"
                rel="noreferrer"
            >
                lav2u
            </a>
            <div>Â© 2023</div>
        </div>
    );
};
