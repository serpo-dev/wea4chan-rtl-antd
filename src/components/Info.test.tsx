import { render, screen } from "@testing-library/react";
import { Info } from "./Info";
import { BrowserRouter } from "react-router-dom";

describe("should render Info properly", () => {
    it("should match the snapshot", () => {
        const { asFragment } = render(<Info />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });

    it("should contain a route to 'about' page", () => {
        render(<Info />, { wrapper: BrowserRouter });
        const about_link: HTMLAnchorElement = screen.getByTestId("about_link");
        expect(about_link.pathname).toEqual("/about");
    });
});
