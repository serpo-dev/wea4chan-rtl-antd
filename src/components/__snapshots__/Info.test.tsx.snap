// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Info properly should match the snapshot 1`] = `
<DocumentFragment>
  <div
    style="font-size: 16px; text-align: center; font-weight: 500; padding: 15px;"
  >
    <a
      data-testid="about_link"
      href="/about"
    >
      About
    </a>
    <span>
       | 
    </span>
    <a
      href="https://github.com/yphwd"
      rel="noreferrer"
      target="_blank"
    >
      lav2u
    </a>
    <div>
      Â© 2023
    </div>
  </div>
</DocumentFragment>
`;
